crates/lib3h/src/transport/transport_encoding/mod.rs

```diff
@@ -0,0 +1,641 @@
+        let (s1out, r1out) = crossbeam_channel::unbounded();
+        let (s1in, r1in) = crossbeam_channel::unbounded();
+
+        // create the first encoding transport
+        let mut t1: TransportActorParentWrapper<()> = GhostParentWrapper::new(
+            Box::new(TransportEncoding::new(
+                crypto.box_clone(),
+                ID_1.to_string(),
+                Box::new(KeystoreStub::new()),
+                Box::new(TransportMock::new(s1out, r1in)),
+            )),
+            "test1",
+        );
+
+        // give it a bind point
+        t1.request(
+            std::time::Duration::from_millis(2000),
+            (),
+            RequestToChild::Bind {
+                spec: Url::parse("test://1").expect("can parse url"),
+            },
+            Box::new(|_, _, response| {
+                assert_eq!(
+                    &format!("{:?}", response),
+                    "Response(Ok(Bind(BindResultData { bound_url: \"test://1/bound?a=HcSCJ9G64XDKYo433rIMm57wfI8Y59Udeb4hkVvQBZdm6bgbJ5Wgs79pBGBcuzz\" })))"
+                );
+                Ok(())
+            })
+        );
+
+        // allow process
+        t1.process(&mut ()).unwrap();
+
+        // we need some channels into our mock inner_transports
+        let (s2out, r2out) = crossbeam_channel::unbounded();
+        let (s2in, r2in) = crossbeam_channel::unbounded();
+
+        // create the second encoding transport
+        let mut t2: TransportActorParentWrapper<()> = GhostParentWrapper::new(
+            Box::new(TransportEncoding::new(
+                crypto.box_clone(),
+                ID_2.to_string(),
+                Box::new(KeystoreStub::new()),
+                Box::new(TransportMock::new(s2out, r2in)),
+            )),
+            "test2",
+        );
+
+        // give it a bind point
+        t2.request(
+            std::time::Duration::from_millis(2000),
+            (),
+            RequestToChild::Bind {
+                spec: Url::parse("test://2").expect("can parse url"),
+            },
+            Box::new(|_, _, response| {
+                assert_eq!(
+                    &format!("{:?}", response),
+                    "Response(Ok(Bind(BindResultData { bound_url: \"test://2/bound?a=HcMCJ8HpYvB4zqic93d3R4DjkVQ4hhbbv9UrZmWXOcn3m7w4O3AIr56JRfrt96r\" })))"
+                );
+                Ok(())
+            })
+        );
+
+        // allow process
+        t2.process(&mut ()).unwrap();
+
+        let mut t1_got_success_resp = false;
+
+        // now we're going to send a message to our sibling #2
+        t1.request(
+            std::time::Duration::from_millis(2000),
+            (),
+            RequestToChild::SendMessage {
+                address: addr2full.clone(),
+                payload: b"hello".to_vec(),
+            },
+            Box::new(|m, _, response| {
+                match m.downcast_mut::<bool>() {
+                    None => panic!("bad downcast"),
```

Comment:
I'm hoping this sort of code is non existent or hidden from a macro in the future.

Reply:
Totally : )